assets/js/components/chess-board-square.js

```diff
@@ -15,28 +15,41 @@ class ChessBoardSquare extends React.Component {
   }
 
   selectSquare() {
-    var { piece, store } = this.props;
-    var { gameId, selectedSquare } = store.getState();
+    let { piece, store } = this.props;
+    let { gameId, selectedSquare, player } = store.getState();
 
-    if (selectedSquare != null) {
+    if (selectedSquare != null && this.moveIsValid()) {
```

Comment:
Would it make sense to move  `selectedSquare != null` into `moveIsValid`? It seems related to being a valid move.